typespec/Applications.Core/environments.tsp

```diff
@@ -103,24 +110,47 @@ model SecretConfig {
   secret?: string;
 }
 
-@doc("The Cloud providers configuration")
+#suppress "@azure-tools/typespec-azure-core/bad-record-type"
+@doc("ProviderConfigProperties specifies provider configuration details needed for recipes.")
+model ProviderConfigProperties extends Record<unknown> {
+  @doc("The secrets for the referenced provider resource.")
+  secrets?: Record<RecipeSecret>;
+}
+
+#suppress "@azure-tools/typespec-azure-core/bad-record-type"
+@doc("EnvironmentVariables describes structure enabling environment variables to be set for recipe execution.")
+model EnvironmentVariables extends Record<unknown> {
+  @doc("The provider secrets passed as environment variables.")
+  secrets?: Record<RecipeSecret>;
+}
+
+@doc("Specifies the secret details for the provider.")
```

Comment:
The comment seems off, should this be updated since the model name has been updated?